src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -176,6 +176,225 @@ void ImageReplayer<I>::BootstrapProgressContext::update_progress(
 }
 
 template <typename I>
+void ImageReplayer<I>::ReplayStatusMaker::send(Context *on_finish) {
+  dout(20) << dendl;
+
+  if (m_on_finish) {
+    dout(20) << "previous request is still in progress, ignoring" << dendl;
+    on_finish->complete(-EAGAIN);
+    return;
+  }
+
+  m_on_finish = on_finish;
+  get_registered_clients();
+}
+
+template <typename I>
+void ImageReplayer<I>::ReplayStatusMaker::get_registered_clients() {
```

Comment:
Ah, yes, I guess this is the result of the test script constantly flushing. The ImageReplayer just calls Journaler::committed() on every entry applied.

BTW, I have updated PR. I am not sure about guarding entries_behind_master against negative values. Outputing 0 in this case may also look confusing. Maybe don't print entries_behind_master=X in this case at all?
Or if master_position less than mirror_position set master_position to mirror?